pkgs/native_assets_builder/test/test_data/native_dynamic_linking_test.dart

```diff
@@ -56,6 +56,7 @@ void main() async {
       File.fromUri(
         buildInputUri,
       ).writeAsStringSync(jsonEncode(inputBuilder.json));
+      outputDirectory = BuildInput(inputBuilder.json).outputDirectory;
```

Comment:
I don't get this conceptually - why do we need to pass the `outputDirectory` to then retrieve the `outputDirectory`?

Reply:
`BuildInput.outputDirectory` now comes up with it's own directory instead of taking the one from the constructor.

I think we should deprecate the parameter in the constructor so that it's clear that that's no longer used.